src/components/IconButton/IconButton.test.tsx

```diff
@@ -0,0 +1,106 @@
+import { render as renderRtl, screen } from '@testing-library/react';
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+
+import { ReactComponent as ExitIcon } from './ExitUnstyled.svg';
+import type { IconButtonProps } from './IconButton';
+import { IconButton, IconButtonVariant, IconButtonSize } from './IconButton';
+
+const render = (props: Partial<IconButtonProps> = {}) => {
+  const allProps = {
+    icon: <ExitIcon />,
+    variant: IconButtonVariant.Secondary,
+    size: IconButtonSize.Small,
+    onClick: jest.fn(),
+    ...props,
+  };
+
+  renderRtl(<IconButton {...allProps} />);
+};
+
+const user = userEvent.setup();
+
+describe('IconButton', () => {
+  Object.values(IconButtonSize).forEach((size) => {
+    it(`should render IconButton with correct classname when variant is ${size}`, () => {
+      render({ size });
+      const otherSizes = Object.values(IconButtonSize).filter((s) => s != size);
+      const iconButtonWrapper = screen.getByTestId('icon-button-wrapper');
+
+      expect(
+        iconButtonWrapper.classList.contains(`icon-button__wrapper--${size}`),
+      ).toBe(true);
+
+      otherSizes.forEach((v) => {
+        expect(
+          iconButtonWrapper.classList.contains(`panel__content-wrapper--${v}`),
```

Comment:
This shouldnt be checking for `panel`. Even if it had the correct classname here, a refactor later could change it to something else and the test still passes. Checking for stuff that shouldnt be there is always a bit problematic, since the test will pass even when there is a typo. That being said, I still think this is a valuable test, at least until we get some screenshot compare tests up and running.

Ensuring that the test actually fails when its supposed to is a general good practice to follow. By "breaking the component" - in this particular case, make sure the component gets the classname it shouldnt so you can see the test actually fail.